// Copyright 2021 The ChromiumOS Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

import "handwriting_interface.proto";

package chrome_knowledge;

// Message type used in web_platform_handwriting_fuzzer. It contains the
// necessary fields for populate input and output.
message WebPlatformHandwritingFuzzContainer {
  // Used for populating a HandwritingModelConstraint (mojom struct).
  optional string constraint_language = 1;

  // Used for populating a HandwritingHints (mojom struct).
  optional uint32 hint_alternatives = 2;
  optional string hint_text_context = 3;

  // Used for populating a vector of HandwritingStroke (mojom struct).
  repeated InkStroke strokes = 4;

  // Used as the output proto of fake handwriting library.
  optional HandwritingRecognizerResult recognizer_result = 5;
}
